runtime/gc_modron_startup/mminit.cpp

```diff
@@ -2853,6 +2853,9 @@ gcInitializeDefaults(J9JavaVM* vm)
 	if (gc_policy_gencon == extensions->configurationOptions._gcPolicy) {
 		/* after we parsed cmd line options, check if we can obey the request to run CS (valid for Gencon only) */
 		if (extensions->concurrentScavengerForced) {
+#if defined(J9VM_ARCH_X86) && defined(J9VM_ENV_DATA64) && defined(J9VM_GC_COMPRESSED_POINTERS)
+			extensions->concurrentScavenger = true;
+#else /* J9VM_ARCH_X86 && J9VM_ENV_DATA64 && J9VM_GC_COMPRESSED_POINTERS */
```

Comment:
it would be simpler code to rely on the branch down below that does:

 						|| (extensions->softwareEvacuateReadBarrier)) {
 					extensions->concurrentScavenger = true;

That would of course require for the two j9sysinfo calls run stable and return false.

Reply:
On X86, no CPU feature check is necessary to enable CS. JIT is designed to support CS on all CPUs that OpenJ9 runs.

Reply:
correct, the CPU check can be skipped (by putting into a separate  GC helper and guard this with something Z specific), but I suspect/hope the check is anyway a no-op (by similar guard on port lib layer).

bottom line I don't like adding another code branch that is X86 specific. CS would be triggered by softwareEvacuateReadBarrier on non h/w assisted platforms. and that branch already exists.

Reply:
Good point. I've updated this logic. X86 now sets `softwareEvacuateReadBarrier` as it is indeed a "software evacuate read barrier"; then we can rely on the branch down below.